harness/determined/common/api/profiler.py

```diff
@@ -46,3 +46,45 @@ def post_trial_profiler_metrics_batches(
         "/api/v1/trials/profiler/metrics",
         body={"batches": [b.__dict__ for b in batches]},
     )
+
+
+class TrialProfilerSeriesLabels:
+    def __init__(self, trial_id: int, name: str, agent_id: str, gpu_uuid: str, metric_type: str):
+        self.trial_id = str(trial_id)
+        self.name = name
+        self.agent_id = agent_id
+        self.gpu_uuid = gpu_uuid if gpu_uuid != "" else None  # type: Optional[str]
+        self.metric_type = metric_type
+
+
+@backoff.on_exception(  # type: ignore
+    backoff.constant,
+    RequestException,
+    max_tries=2,
+    giveup=lambda e: e.response is not None and e.response.status_code < 500,
+)
+def get_trial_profiler_available_series(
+    master_url: str,
+    trial_id: str,
+) -> List[TrialProfilerSeriesLabels]:
+    """
+    Get available profiler series for a trial. This uses the non-streaming version of the API
+    """
+    follow = False
```

Comment:
maybe just make this an optional param?

Reply:
I hardcoded it to `False` because I think the interface for a streaming API should probably be streaming-centric, but I'm not sure exactly what it should be (maybe it returns a generator?). That problem is best solved when someone has a use case for it..